eclair-core/src/main/scala/fr/acinq/eclair/wire/ChannelCodecs.scala

```diff
@@ -363,16 +323,25 @@ object ChannelCodecs extends Logging {
    *
    * More info here: https://github.com/scodec/scodec/issues/122
    */
-  val stateDataCodec: Codec[HasCommitments] = ("version" | constant(0x00)) ~> discriminated[HasCommitments].by(uint16)
-    .typecase(0x10, DATA_NORMAL_Codec)
-    .typecase(0x09, DATA_CLOSING_Codec)
-    .typecase(0x08, DATA_WAIT_FOR_FUNDING_CONFIRMED_Codec)
-    .typecase(0x01, DATA_WAIT_FOR_FUNDING_CONFIRMED_COMPAT_01_Codec)
-    .typecase(0x02, DATA_WAIT_FOR_FUNDING_LOCKED_Codec)
-    .typecase(0x03, DATA_NORMAL_COMPAT_03_Codec)
-    .typecase(0x04, DATA_SHUTDOWN_Codec)
-    .typecase(0x05, DATA_NEGOTIATING_Codec)
-    .typecase(0x06, DATA_CLOSING_COMPAT_06_Codec)
-    .typecase(0x07, DATA_WAIT_FOR_REMOTE_PUBLISH_FUTURE_COMMITMENT_Codec)
+  val stateDataCodec: Codec[HasCommitments] = discriminated[HasCommitments].by(byte)
+    .typecase(1, discriminated[HasCommitments].by(uint16)
```

Comment:
I've set a new version, wasn't really mandatory. We're relying on the same ordering trick (see comment) to take care of the migration.

Reply:
Nice, that means we write with version 1, and legacy will fail the read with version 1 and fallback to version 0, that sounds good.

Reply:
It would have fallen back using the 2nd-level typecase anyway, the change of version is strictly non-necessary.

Reply:
Got it, indeed since the typecase are different it works without the version.
But it clearly separates the two groups, I think it makes it clearer that it's a milestone/major change.